cmd/prom-migrator/main.go

```diff
@@ -106,6 +108,18 @@ func main() {
 		os.Exit(0)
 	}
 
+	ms, err := parser.ParseExpr(conf.readerMetricsMatcher)
+	if err != nil {
+		log.Error("msg", "could not parse reader metrics matcher", "error", err)
+		os.Exit(2)
+	}
+
+	vs, ok := ms.(*parser.VectorSelector)
```

Comment:
lets shift this check to `func validateConf(conf *config) error` same as the above comment, and change the log to return an error as per the `validate()`'s purpose. The error can be like `fmt.Errorf("'-reader-metrics-matcher' should always be a valid vector_selector as per PromQL")`.